### PR TITLE
Add realistic example

### PR DESCRIPTION
**Pull request by ilopezluna**
In this PR we changed the way how we were using Testcontainers to have a more realistic example:

- Using [singleton container pattern](https://www.testcontainers.org/test_framework_integration/manual_lifecycle_control/#singleton-containers) to start a Kafka container only once
- Also defined `sourceCompatibility` & `targetCompatibility` to use Java 8. This way the project can be used by more people. WDYT?

Running tests in a M1 Pro take:

- ~34s with Turbo Mode
- ~43s without Turbo Mode
- More than 1 minute without Turbo Mode & without forks

### CODE REVIEW

File: src/test/java/org/testcontainers/cloud/turbomode/sample/AbstractIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        }
+
+        // use consumer to check received records
+        KafkaConsumer<String, String> consumer = createConsumer(topic);
+
+        AtomicInteger readMessages = new AtomicInteger();
+        await().atMost(Duration.TWO_MINUTES).untilAsserted(() -> {
+            ConsumerRecords<String, String> records = consumer.poll(100);
+            readMessages.addAndGet(records.count());
+            assertEquals(readMessages.get(), messageCount);
```
Comment (ilopezluna) [inline src/test/java/org/testcontainers/cloud/turbomode/sample/AbstractIntegrationTest.java:55]: @kiview @clive-jevons the test were failing because we were writing in the same topic with another test, the `EventWriterJupiterTests` that we created, remember? After removing it, tests run in all 3 scenarios. So I think it's not needed to remove the already read messages [as here](https://github.com/AtomicJar/testcontainers-cloud-demo/blob/master/kafka-mongodb-repro/src/test/java/cloud/testcontainers/PugPartyInviteTest.java). WDYT?

Comment (kiview) [reply src/test/java/org/testcontainers/cloud/turbomode/sample/AbstractIntegrationTest.java:55]: Oh right, no need for the `EventWriterJupiterTests`. In this case, fine with me 👍

File: build.gradle [hunk 2]
```diff
@@ -4,19 +4,24 @@ plugins {
 
 group 'org.testcontainers.cloud.turbomode.sample'
 version '1.0-SNAPSHOT'
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
```
Comment (kiview) [inline build.gradle:8]: While we are at it, we can also add a JVM toolchain, then users don't even need an appropriate JDK:
https://docs.gradle.org/current/userguide/toolchains.html

I know there were issues around JDK8 for M1 though, @eddumelendez might know more. However, if we pull in the appropriate JDK via toolchain, it is also fine to user a more modern LTS version.

File: build.gradle [hunk 3]
```diff
@@ -4,19 +4,24 @@ plugins {
...
     testImplementation 'org.junit.jupiter:junit-jupiter:5.9.0'
-    testImplementation platform('org.testcontainers:testcontainers-bom:1.17.4')
+    testImplementation platform('org.testcontainers:testcontainers-bom:1.17.5')
     testImplementation "org.testcontainers:testcontainers"
     testImplementation "org.testcontainers:junit-jupiter"
     testImplementation 'ch.qos.logback:logback-classic:1.3.3'
     testImplementation 'org.assertj:assertj-core:3.23.1'
-    testImplementation "org.testcontainers:kafka:1.17.4"
+    testImplementation "org.testcontainers:kafka"
+    testImplementation 'org.testcontainers:kafka'
```
Comment (kiview) [inline build.gradle:22]:
```suggestion
```

Duplicate

File: build.gradle [hunk 4]
```diff
@@ -4,19 +4,24 @@ plugins {
...
+    implementation 'org.apache.kafka:kafka-clients:1.0.0'
+    testImplementation group: 'org.awaitility', name: 'awaitility', version: '3.1.6'
```
Comment (kiview) [inline build.gradle:24]:
```suggestion
    testImplementation 'org.awaitility:awaitility:3.1.6'
```
Short-form everywhere

File: src/test/java/org/testcontainers/cloud/turbomode/sample/AbstractIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+        KafkaConsumer<String, String> consumer = new KafkaConsumer<>(kafkaConfig);
+        consumer.subscribe(Collections.singletonList(topic));
+        return consumer;
+    }
+
+    protected void writeAndVerify(String clientId, String topic) {
+        EventWriter notifier = new EventWriter(kafka.getBootstrapServers(), clientId, topic);
+        String notificationMessage = "foobar";
+        int messageCount = 100000;
```
Comment (kiview) [inline src/test/java/org/testcontainers/cloud/turbomode/sample/AbstractIntegrationTest.java:43]: Not directly necessary, but it is a little bit nicer if we make `messageCount` a method parameter.

File: src/test/java/org/testcontainers/cloud/turbomode/sample/TestcontainersCloud1Test.java [hunk 6]
```diff
@@ -1,20 +1,13 @@
...
-import org.testcontainers.junit.jupiter.Container;
 import org.testcontainers.junit.jupiter.Testcontainers;
-import org.testcontainers.utility.DockerImageName;
 
 @Testcontainers
-public class TestcontainersCloud1Test {
-
-    @Container
-    private KafkaContainer kafka = new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka:6.2.1"));
+public class TestcontainersCloud1Test extends AbstractIntegrationTest {
```
Comment (kiview) [inline src/test/java/org/testcontainers/cloud/turbomode/sample/TestcontainersCloud1Test.java:7]: The `@Testcontainers` annotation can be removed from all concrete test classes.

File: build.gradle [hunk 7]
```diff
@@ -4,19 +4,24 @@ plugins {
...
 repositories {
     mavenCentral()
 }
 
 dependencies {
     testImplementation 'org.junit.jupiter:junit-jupiter:5.9.0'
-    testImplementation platform('org.testcontainers:testcontainers-bom:1.17.4')
+    testImplementation platform('org.testcontainers:testcontainers-bom:1.17.5')
     testImplementation "org.testcontainers:testcontainers"
     testImplementation "org.testcontainers:junit-jupiter"
```
Comment (kiview) [inline build.gradle:18]:
```suggestion
```
Not used anymore.

File: build.gradle [hunk 8]
```diff
@@ -4,19 +4,24 @@ plugins {
...
 
 repositories {
     mavenCentral()
 }
 
 dependencies {
     testImplementation 'org.junit.jupiter:junit-jupiter:5.9.0'
-    testImplementation platform('org.testcontainers:testcontainers-bom:1.17.4')
+    testImplementation platform('org.testcontainers:testcontainers-bom:1.17.5')
     testImplementation "org.testcontainers:testcontainers"
```
Comment (kiview) [inline build.gradle:17]:
```suggestion
```
Not necessary, since transitive dependency from `kafka`.

File: build.gradle [hunk 9]
```diff
@@ -4,19 +4,24 @@ plugins {
...
     testImplementation "org.testcontainers:junit-jupiter"
     testImplementation 'ch.qos.logback:logback-classic:1.3.3'
     testImplementation 'org.assertj:assertj-core:3.23.1'
```
Comment (kiview) [inline build.gradle:20]: Looks like it is not used.

**Review by kiview — APPROVED** [review]
See my review comments, besides I think it is a considerable improvement, so apply suggestion as you see fit 👍

File: build.gradle (hunk 9 above)
Comment (ilopezluna) [reply build.gradle]: It was used in `TestcontainersCloudFirstTest` but I changed to use assertions from jupiter as well, this way we can remove the dependency

File: src/test/java/org/testcontainers/cloud/turbomode/sample/AbstractIntegrationTest.java (hunk 1 above)
Comment (clive-jevons) [reply src/test/java/org/testcontainers/cloud/turbomode/sample/AbstractIntegrationTest.java:55]: If we're not using "exactly once semantics" we run the risk of flaky tests if we don't use the "track UUID per message" approach. So we _could_ conceivably run into the case where for a user the tests fail - not because their setup is bad, but because they've run into the edge case where the message gets received twice.

But - it is also entirely legitimate to say that we'll wait until we've hit that problem before fixing it 😂 
I'll leave it up to you @ilopezluna

File: src/test/java/org/testcontainers/cloud/turbomode/sample/TestcontainersCloud1Test.java [hunk 10]
```diff
@@ -1,20 +1,11 @@
...
 
     @Test
-    public void testFindingAnInsertedValue() {
-        Assertions.assertNotNull(kafka.getBootstrapServers());
+    public void writeAndVerify() {
```
Comment (clive-jevons) [inline src/test/java/org/testcontainers/cloud/turbomode/sample/TestcontainersCloud1Test.java:8]: Absolute wincy-tincy-minor nit: my understanding is that the preferred syntax for tests using the new junit is to leave the modifier (`public` in this case) out completely (so just `void writeAndVerify()` here? But obviously it works as is, so entirely optional.

**Review by clive-jevons — APPROVED** [review]
approved with insignificantly minor nits 😬

File: build.gradle (hunk 2 above)
Comment (eddumelendez) [reply build.gradle]: the only vendors providing Java 8 for M1 are bellsoft, corretto and zulu. I know that Temurin is planning to release it this month, hopefully.
